pkg/ccl/backupccl/file_sst_sink.go

```diff
@@ -264,21 +268,27 @@ func (s *fileSSTSink) write(ctx context.Context, resp exportedSpan) (roachpb.Key
 		return nil, err
 	}
 
+	var latestSpanMidRow bool
 	if len(resp.resumeKey) > 0 {
-		span.EndKey, s.midRow = adjustFileEndKey(span.EndKey, maxKey, maxRange)
+		span.EndKey, latestSpanMidRow = adjustFileEndKey(span.EndKey, maxKey, maxRange)
 		// Update the resume key to be the adjusted end key so that start key of the
 		// next file is also clean.
 		resp.resumeKey = span.EndKey
-	} else {
-		s.midRow = false
 	}
 
+	safeSizeBasedFileCut := !s.midRow &&
+		len(s.flushedFiles) > 0 &&
+		s.flushedFiles[len(s.flushedFiles)-1].EntryCounts.DataSize >= logicalBackupFileSize
+
 	// If this span extended the last span added -- that is, picked up where it
 	// ended and has the same time-bounds -- then we can simply extend that span
 	// and add to its entry counts. Otherwise we need to record it separately.
+	// Further, if latest file is sufficiently large, create a new span if it is
```

Comment:
minor nit: to me, conceptually, _extending_ a span is the thing that requires lots of eligibility conditions be met, including that the span isn't too big, with making a new span being the default, rather than the span being "big enough" being something that allows not-extending?

Reply:
Although it is weird that if we're mid-key, we _must_ extend, but you would not know that reading this conditional before your change; extending is seen as something we _might_ do, not _must do if mid. 

I'd maybe take this chance to change this up a bit, to make midRow's role more obvious and also not denormalize some stuff into `safeSizeBasedFileCut` that really belongs here / not duplicated. e.g:
```
extend := false
if l := len(s.flushedFiles); l > 0 && ! s.midRow {
last := s.flushedFiles[l-1]
extend = last.EndTime(...) && ..&& last.EntryCounts.DataSize < fileSpanByteLimit
}
if extend {
...
} else {
...
}
```